cfgov/v1/tests/test_links.py

```diff
@@ -22,6 +26,41 @@ def test_gov_link(self):
         self.assertIn('class="icon-link_text"', output)
 
 
+class LinkTest(TestCase):
+
+    def setUp(self):
+
+        self.page = mock.MagicMock()
+        self.request = mock.MagicMock()
+        self.request.url = 'http://localhost:8000/about-us/blog/'
+        self.page.url = 'http://localhost:8000/about-us/blog/we-want-hear-public-about-payday-loans/'
+        self.page.live = True
+        self.page.specific.shared = True
+        self.context = mock.MagicMock()
+
+        def context_dict(name):
+            return {'request': self.request}[name]
+
+        self.context.__getitem__.side_effect = context_dict
+
+    def test_get_protected_url(self):
+
+        result = get_protected_url(self.context, None)
+        self.assertEqual(result, '#')
+        self.page.url = None
+        result = get_protected_url(self.context, self.page)
+        self.assertIs(result, None)
```

Comment:
probably depends on your ide.  i use pycharm.

Here's the list of assert methods available in unittest:

https://docs.python.org/2/library/unittest.html#assert-methods